core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -1075,7 +1075,11 @@ private void setNamesIfAbsent() {
                 }
             }
 
-            serviceName = newServiceName;
+            if (config != null) {
+                serviceName = config.server().config().serviceNaming().convert(newServiceName);
+            } else {
+                serviceName = newServiceName;
+            }
```

Comment:
I'm not sure if this is the right place to name a service. How about making `ServiceConfig` call `ServiceNamingRule.convert()` at its construction time, so that `ServiceConfig.defaultServiceName()` is determined there?

Reply:
> construction time

Basically, that sounds good to me when `defaultServiceName` is given. However, how do we convert service names when `defaultServiceName` is not given? In case of `DefaultRequestLog`, a service name is determined [from a service type or innermost class name](https://github.com/line/armeria/blob/armeria-1.5.0/core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java#L1050-L1068). Thus, for example, utility services such as `HealthCheckService` or `DocService` will be out of affected scope.

Reply:
Because `ServiceNaming.apply()` will be invoked with `ServiceConfig`, you can get the innermost class name from `ServiceConfig.service()` after unwrapping all decorators. Makes sense?

Reply:
i.e.

```java
public final class ServiceConfig {
    ServiceConfig(..., HttpService service, ..., ServiceNaming serviceNaming, ...) {
        ...
        defaultServiceName = serviceNaming.apply(this);
    }
}
```

Reply:
Makes sense. I concerned how to set `defaultServiceNaming` to utility services. 

> How about adding per-route service naming **as well**?

Related to that, I have a question between per-route service naming and global service naming. per-route service naming is supposed to be prioritized. Thus, when building `ServiceConfig`, `ServerBuilder#serviceNaming` is expected to be used if per-routing service naming is not given. For this, builder pattern may be complicated to propagate the property. Does it make sense?

Reply:
Correct. Most of property priority implementation resides in `VirtualHostBuilder.build(VirtualHostBuilder template)`, so it shouldn't be that difficult once you take a look. :wink: